Userland/Libraries/LibWeb/DOM/Element.cpp

```diff
@@ -167,13 +167,12 @@ JS::GCPtr<Attr> Element::get_attribute_node_ns(Optional<FlyString> const& namesp
 WebIDL::ExceptionOr<void> Element::set_attribute(FlyString const& name, String const& value)
 {
     // 1. If qualifiedName does not match the Name production in XML, then throw an "InvalidCharacterError" DOMException.
-    // FIXME: Proper name validation
     if (name.is_empty())
         return WebIDL::InvalidCharacterError::create(realm(), "Attribute name must not be empty"_fly_string);
+    TRY(Document::validate_qualified_name(realm(), name));
```

Comment:
I don't think this is the right validation method - the step here says to validate against the `Name` production, which is https://www.w3.org/TR/xml/#NT-Name

`Document::validate_qualified_name` (https://dom.spec.whatwg.org/#validate) instead validates against the `QName` production, which is: https://www.w3.org/TR/xml-names/#NT-QName

Reply:
Thanks for the catch! There looks to be an existing validation for name, so I used that. I did however need to change the parameter to FlyString rather than String. It didn't run into errors while building so I hope that is sufficient.

Reply:
Great!

However, instead of changing that parameter to `FlyString`, let's change the call site in `Element::set_attribute` to pass in `String`. So that function invocation would look something like like:

```c++
if (!Document::is_valid_name(name.to_string()))
    return WebIDL::InvalidCharacterError::create(realm(), "Attribute name must not be empty or contain invalid characters"_fly_string);
```

The reason to avoid changing `Document::is_valid_name` to accept a `FlyString` is that other call sites that pass in a `String` will now implicitly do a `FlyString` construction from that string. And that isn't quite free, it involves e.g. doing a hash table lookup to see if that string exists. Whereas extracting the underlying `String` from `FlyString::to_string` is much cheaper.

(For background on what `FlyString` even is, it's an implementation of the [flyweight pattern](https://en.wikipedia.org/wiki/Flyweight_pattern) for strings. It basically serves to make sure there's only a single instance of a string in memory, and each `FlyString` instance basically points to that single instance.)